### DISCUSSION TITLE
JWT_SESSION_ERROR decryption operation failed

### DISCUSSION THREAD

**Discussion by ghost** [nextauthjs/next-auth#8370] [Help]
### Environment

System:
    OS: Linux 5.15 Alpine Linux
    CPU: (2) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
    Memory: 1.89 GB / 3.83 GB
    Container: Yes
    Shell: 1.35.0 - /bin/ash
  Binaries:
    Node: 18.15.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.5.0 - /usr/local/bin/npm

Next-auth 4.23.1

### Reproduction URL

/

### Describe the issue

When using the JWT Callback, and populate the session (using cookies) with some data that makes the tokens split into 2 tokens, the callback throw an error : 

```
  callbacks: {
    jwt: async ({ token, user, account }) => {
      // New session
      if (user && account) {
        return Promise.resolve({
          accessToken: user.accessToken,
          refreshToken: user.refreshToken
          // ...token
        })
      }

      return Promise.resolve({
        ...token,
        accessToken: token.accessToken
      })
    }
  }
```

```
[next-auth][error][JWT_SESSION_ERROR] 
https://next-auth.js.org/errors#jwt_session_error decryption operation failed {
  message: 'decryption operation failed',
  stack: 'JWEDecryptionFailed: decryption operation failed\n' +
    '    at gcmDecrypt (/srv/app/node_modules/jose/dist/node/cjs/runtime/decrypt.js:67:15)\n' +
    '    at decrypt (/srv/app/node_modules/jose/dist/node/cjs/runtime/decrypt.js:92:20)\n' +
    '    at flattenedDecrypt (/srv/app/node_modules/jose/dist/node/cjs/jwe/flattened/decrypt.js:119:52)\n' +
    '    at async compactDecrypt (/srv/app/node_modules/jose/dist/node/cjs/jwe/compact/decrypt.js:18:23)\n' +
    '    at async jwtDecrypt (/srv/app/node_modules/jose/dist/node/cjs/jwt/decrypt.js:8:23)\n' +
    '    at async Object.decode (/srv/app/node_modules/next-auth/jwt/index.js:64:7)\n' +
    '    at async Object.session (/srv/app/node_modules/next-auth/core/routes/session.js:43:28)\n' +
    '    at async AuthHandler (/srv/app/node_modules/next-auth/core/index.js:165:27)\n' +
    '    at async file:///srv/app/.nuxt/dev/index.mjs:960:24\n' +
    '    at async Object.handler (file:///srv/app/node_modules/h3/dist/index.mjs:1284:19)',
  name: 'JWEDecryptionFailed'
}
```

![image](https://github.com/nextauthjs/next-auth/assets/65252347/368a7213-efd5-421c-a2f5-97892fabe9b9)

But, when populating the session with less data : 

```
callbacks: {
    jwt: async ({ token, user, account }) => {
      // New session
      if (user && account) {
        return Promise.resolve({
          accessToken: user.accessToken
          // refreshToken: user.refreshToken
          // ...token
        })
      }

      return Promise.resolve({
        ...token,
        accessToken: token.accessToken
      })
    }
  }
```

![image](https://github.com/nextauthjs/next-auth/assets/65252347/a35ed79c-f22e-4948-92de-f430f490edaa)

No errors...

### How to reproduce

Create next-auth project, add some codes to auth user, and add JWT callback

### Expected behavior

No errors...

Tell me if it is really necessary to add repository

**Comment by douglasrcjames** [comment]
I am also suddenly getting this error, however, I do not see it affecting the functionality of my app. Seems to be connected to a recent Next.js update or NextAuth update. 

My current versions are:
```
    "next": "^13.4.17",
    "next-auth": "^4.23.1",
```

**Reply by ghost** [reply]
I'm using Nuxt (not Next) sidebase auth package so I don't think it's related to Next directly.
recently, they commited something about token chunks https://github.com/nextauthjs/next-auth/pull/8284

**Comment by oviava** [comment]
I'm having this issue as well, weirdly enough I'm not having it running it locally but occurs when deployed. 

Using JWT session

**Reply by oviava** [reply]
The issue seems to be from the 'value' getter on Session store

``` ts
get value() {
  // Sort the chunks by their keys before joining
  const sortedKeys = Object.keys(this.#chunks).sort((a, b) => {
    const aSuffix = parseInt(a.split(".").pop() || "0")
    const bSuffix = parseInt(b.split(".").pop() || "0")

    return aSuffix - bSuffix
  });

  // Use the sorted keys to join the chunks in the correct order
  return sortedKeys.map(key => this.#chunks[key]).join("")
}

```

It only happens when there are 2 token cookies generated

**Comment by emreakdas** [comment]
I have the same problem, the session drops suddenly when using OIDC. @balazsorban44
